common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java

```diff
@@ -0,0 +1,327 @@
+        KubeClusterException error = null;
+        for (File f : files) {
+            if (f.isFile()) {
+                if (f.getName().endsWith(".yaml")) {
+                    try {
+                        Exec.exec(namespacedCommand(subcommand, "-f", f.getAbsolutePath()));
+                    } catch (KubeClusterException e) {
+                        if (error == null) {
+                            error = e;
+                        }
+                    }
+                }
+            } else if (f.isDirectory()) {
+                File[] children = f.listFiles();
+                if (children != null) {
+                    Arrays.sort(children, cmp);
+                    KubeClusterException e = execRecursive(subcommand, children, cmp);
+                    if (error == null) {
+                        error = e;
+                    }
+                }
+            } else if (!f.exists()) {
+                throw new RuntimeException(new NoSuchFileException(f.getPath()));
+            }
+        }
+        return error;
+    }
+
+    @Override
+    public K replace(File... files) {
+        try (Context context = defaultContext()) {
+            execRecursive("replace", files, (f1, f2) -> f1.getName().compareTo(f2.getName()));
+            return (K) this;
+        }
+    }
+
+    @Override
+    public K replaceContent(String yamlContent) {
+        try (Context context = defaultContext()) {
+            Exec.exec(yamlContent, namespacedCommand("replace", "-f", "-"));
+            return (K) this;
+        }
+    }
+
+    @Override
+    public K createContent(String yamlContent) {
+        try (Context context = defaultContext()) {
+            Exec.exec(yamlContent, namespacedCommand(CREATE, "-f", "-"));
+            return (K) this;
+        }
+    }
+
+    @Override
+    public K deleteContent(String yamlContent) {
+        try (Context context = defaultContext()) {
+            Exec.exec(yamlContent, namespacedCommand(DELETE, "-f", "-"));
+            return (K) this;
+        }
+    }
+
+    @Override
+    public K createNamespace(String name) {
+        try (Context context = adminContext()) {
+            Exec.exec(namespacedCommand(CREATE, "namespace", name));
+        }
+        return (K) this;
+    }
+
+    @Override
+    public K deleteNamespace(String name) {
+        try (Context context = adminContext()) {
+            Exec.exec(namespacedCommand(DELETE, "namespace", name));
+        }
+        return (K) this;
+    }
+
+    @Override
+    public ProcessResult exec(String pod, String... command) {
+        List<String> cmd = namespacedCommand("exec", pod, "--");
+        cmd.addAll(asList(command));
+        return Exec.exec(cmd);
+    }
+
+    enum ExType {
+        BREAK,
+        CONTINUE,
+        THROW
+    }
+
+    private K waitFor(String resource, String name, Predicate<JsonNode> ready) {
+        long timeoutMs = 240_000L;
```

Comment:
It seems that this timeout it's not enough. On a fresh cluster, it took more the 4 minutes for pulling down the Kafka Docker image and the test failed.